packages/extension/src/view/devtools/pages/layout.tsx

```diff
@@ -202,23 +200,111 @@ const Layout = ({ setSidebarData }: LayoutProps) => {
 
   const buttonReloadActionCompnent = useMemo(() => {
     return (
-      <Button text={'Reload'} onClick={handleSettingsChange} variant="large" />
+      <div className="flex items-center gap-5">
+        <Button
+          text="Yes"
+          size="large"
+          onClick={() => {
+            handleSettingsChange();
+          }}
+          variant="success"
+        />
+        <Button
+          text="Cancel"
+          size="large"
+          onClick={async () => {
+            await chrome.storage.session.remove([
+              'isUsingCDP',
+              'pendingReload',
+            ]);
+            setSettingsChanged(false);
+            setIsUsingCDPForSettingsPageDisplay(true);
+          }}
+        />
+      </div>
     );
-  }, [handleSettingsChange]);
+  }, [
+    handleSettingsChange,
+    setIsUsingCDPForSettingsPageDisplay,
+    setSettingsChanged,
+  ]);
+
+  const isUsingCDPCondition = useMemo(() => {
+    if (isUsingCDPForSettingsPageDisplay) {
+      return true;
+    } else {
+      return false;
+    }
+  }, [isUsingCDPForSettingsPageDisplay]);
 
   const settingsReadActionComponent = useMemo(() => {
     return (
-      <div
-        className="w-14 h-14 flex items-center"
-        onClick={() => {
-          chrome.storage.session.set({ readSettings: true });
-          setHasWarningBeenShown(true);
-        }}
-      >
-        <PaddedCross className="w-4 h-4" />
+      <div className="flex items-center gap-5">
+        <Button
+          text="Yes"
+          size="large"
+          onClick={() => {
+            handleSettingsChange();
+          }}
+          variant={exceedingLimitations ? 'danger' : 'success'}
+        />
+        <Button
+          text="Cancel"
+          size="large"
+          onClick={async () => {
+            await chrome.storage.session.remove([
+              'isUsingCDP',
+              'pendingReload',
+            ]);
+            setSettingsChanged(false);
+            setIsUsingCDPForSettingsPageDisplay(false);
+          }}
+        />
       </div>
     );
-  }, [setHasWarningBeenShown]);
+  }, [
+    exceedingLimitations,
+    handleSettingsChange,
+    setSettingsChanged,
+    setIsUsingCDPForSettingsPageDisplay,
+  ]);
+
+  const formedToastMessage = useMemo(() => {
+    let message = '';
+
+    if (settingsChanged) {
+      if (isUsingCDPCondition) {
+        message = exceedingLimitations
+          ? 'Enabling CDP with more than 5 tabs open will impact your browser performance and all tabs will be reloaded. Are you sure you want to enable CDP?'
+          : 'Settings changed, reload all tabs to apply changes?';
```

Comment:
Let's make these constants or something so it doesn't repeat in poup and devtool.